include/zephyr/devicetree.h

```diff
@@ -3676,6 +3704,14 @@
  */
 #define DT_INST_REG_ADDR(inst) DT_INST_REG_ADDR_BY_IDX(inst, 0)
 
+/**
+ * @brief 64-bit version of a DT_INST_REG_ADDR macro.
+ *        Use it in cases when 64-bit address value may appear in Device Tree
+ * @param inst instance number
+ * @return instance's register block address
+ */
+#define DT_INST_REG_ADDR_U64(inst) UINT64_C(DT_INST_REG_ADDR(inst))
```

Comment:
this is missing tests, also, `#include <stdint.h>` is missing

Reply:
I've made tests using the platform which is not currently upstreamed (we are paving the road for this).
Is it ok, or would you suggest some specific tests here?

Reply:
As for stdint, I believe it is getting included through #include <zephyr/sys/util.h> -> #include <zephyr/types.h>

Reply:
All macros in zephyr/devicetree.h (and its sub-headers) require test cases in tests/lib/devicetree. Usually tests/lib/devicetree/api/src/main.c.

Reply:
> As for stdint, I believe it is getting included through #include <zephyr/sys/util.h> -> #include <zephyr/types.h>

Relying on indirect includes is a fragile approach.

Reply:
Agreed, please include what you use (and remember to be careful to allow this header to be included from linker scripts of course).

Reply:
> Relying on indirect includes is a fragile approach.
Agree, will add it thanks!